alot/commands/globals.py

```diff
@@ -857,8 +857,14 @@ def apply(self, ui):
 
         # Figure out whether we should GPG sign messages by default
         # and look up key if so
-        self.envelope.sign = account.sign_by_default
-        self.envelope.sign_key = account.gpg_key
+        if account.sign_by_default:
+            if account.gpg_key:
+                self.envelope.sign = account.sign_by_default
+                self.envelope.sign_key = account.gpg_key
+            else:
+                msg = 'Cannot find gpg key for account {}'.format(account.address)
+                logging.warn(msg)
+                ui.notify(msg, priority='error')
```

Comment:
This change looks good. It also makes `sign_by_default` work like `:sign` which will fail if no key is provided as argument or as the accounts `gpg_key` option.

I think I also found out why `sign_by_default` did previously work when `account.gpg_key = None`: The attribute `envelope.sign_key` is not touched at all until the `:send` command calls [`envelope.construct_mail()`](https://github.com/pazz/alot/blob/4f78529875c2687e58586c43512a71787f72fccb/alot/db/envelope.py#L169-L195) which just passes the key to [`detached_signature_for()`](https://github.com/pazz/alot/blob/4f78529875c2687e58586c43512a71787f72fccb/alot/crypto.py#L150-L170). Now if the key is `None` in there I assume that gpg will just pick up the default key from the gpg config file.